tests/common/member_test.go

```diff
@@ -51,3 +57,56 @@ func TestMemberList(t *testing.T) {
 		})
 	}
 }
+
+func TestMemberAdd(t *testing.T) {
+	testRunner.BeforeTest(t)
+
+	tcs := []struct {
+		name    string
+		learner bool
+	}{
+		{
+			name:    "NotLearner",
+			learner: false,
+		},
+		{
+			name:    "Learner",
+			learner: true,
+		},
+	}
+
+	for _, tc := range tcs {
+		for _, nc := range clusterTestCases {
+			t.Run(tc.name+"/"+nc.name, func(t *testing.T) {
+				ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+				defer cancel()
+				clus := testRunner.NewCluster(ctx, t, nc.config)
+				defer clus.Close()
+				cc := clus.Client()
+
+				testutils.ExecuteUntil(ctx, t, func() {
+					clus.WaitLeader(t)
+					var addResp *clientv3.MemberAddResponse
+					var err error
+					if tc.learner {
+						addResp, err = cc.MemberAddAsLearner("newmember", []string{blackHolePeerUrl})
```

Comment:
Why not to use a real peerURL something like `http://localhost:xxx `instead of `http://240.0.0.0:65535`?

Reply:
`http://240.0.0.0:65535` is used to avoid "Peer URLs already exists" error. If `http://localhost:xxx` happens to be the peer-url of a started etcd instance, `member add` operation would fail due to that error. I considered to use `http://localhost:xxx` with a unused random port such as 2777 and 15981, but I'm slightly worried about port conflict in the future.